torch/distributions/transforms.py

```diff
@@ -652,9 +657,9 @@ def log_abs_det_jacobian(self, x, y):
         for trans, length in zip(self.transforms, self.lengths):
             xslice = x.narrow(self.dim, start, length)
             yslice = y.narrow(self.dim, start, length)
-            logdetjacs.append(trans.log_abs_det_jacobian(xslice, yslice))
+            logdetjacs.append(trans.log_abs_det_jacobian(xslice, yslice).view(x.shape + (1,) * self.event_dim))
```

Comment:
`.view()` --> `.reshape()`

Reply:
What's the benefit of using `.reshape()` over `.view()`?

Reply:
In this case they are identical, but we try to keep `.view()` out of the distributions codebase because it does not catch errors due to non-contiguous layout from upstream transposes.